src/init.cpp

```diff
@@ -660,7 +660,8 @@ void InitParameterInteraction(ArgsManager& args)
             LogPrintf("%s: parameter interaction: -connect set -> setting -listen=0\n", __func__);
     }
 
-    if (args.IsArgSet("-proxy")) {
+    std::string proxy_arg = args.GetArg("-proxy", "");
+    if (proxy_arg != "" && proxy_arg != "0") {
```

Comment:
In commit "init: Prevent -noproxy and -proxy=0 settings from interacting with other settings" (3429d67014095b42a976d95c3ef8622d5fe085e6)

These lines are copied from `AppInitMain` below

https://github.com/bitcoin/bitcoin/blob/2b5a741e98f186e50d9fbe1ceadcc8b8c91547f7/src/init.cpp#L1330-L1331